torch_geometric/explain/algorithm/gnn_explainer.py

```diff
@@ -0,0 +1,550 @@
+                self.subgraph(model, index, x, edge_index, **kwargs)
+            if target_index is not None and model_config.mode ==\
+                    ModelMode.classification:
+                target = torch.index_select(target, 1, subset)
+            else:
+                target = target[subset]
+        elif model_config.task_level == ModelTaskLevel.graph:
+            hard_edge_mask = None
+            subset = None
+        else:
+            raise NotImplementedError
+
+        self._train_node_edge_mask(model, x, edge_index, explainer_config,
+                                   model_config, target, index, target_index,
+                                   **kwargs)
+
+        node_mask, node_feat_mask, edge_mask = self._get_masks(
+            num_nodes=num_nodes, num_edges=num_edges,
+            explainer_config=explainer_config,
+            task_level=model_config.task_level, hard_edge_mask=hard_edge_mask,
+            subset=subset)
+
+        self._clean_model(model)
+
+        # build explanation
+        return Explanation(x=x, edge_index=edge_index, edge_mask=edge_mask,
+                           node_mask=node_mask, node_feat_mask=node_feat_mask)
+
+    def _get_masks(
+        self,
+        num_nodes: int,
+        num_edges: int,
+        explainer_config: ExplainerConfig,
+        task_level: ModelTaskLevel,
+        hard_edge_mask: Optional[Tensor],
+        subset: Optional[Tensor],
+    ) -> Tuple[Optional[Tensor], Optional[Tensor], Optional[Tensor]]:
+        """Extract and reshape the masks from the model parameters."""
+        node_mask_, node_feat_mask_, edge_mask_ = None, None, None
+
+        if self.node_mask is not None:
+            node_mask = self.node_mask.detach().sigmoid().squeeze(-1)
+            if explainer_config.node_mask_type == MaskType.object:
+                if task_level == ModelTaskLevel.node:
+                    node_mask_ = torch.zeros(num_nodes, dtype=torch.float)
+                    node_mask_[subset] = node_mask
+                elif task_level == ModelTaskLevel.graph:
+                    node_mask_ = node_mask
+                else:
+                    raise NotImplementedError
+            elif explainer_config.node_mask_type == MaskType.attributes:
+                if task_level == ModelTaskLevel.node:
+                    node_feat_mask_ = torch.zeros(num_nodes,
+                                                  node_mask.size(-1),
+                                                  dtype=torch.float)
+                    node_feat_mask_[subset] = node_mask
+                elif task_level == ModelTaskLevel.graph:
+                    node_feat_mask_ = node_mask
+                else:
+                    raise NotImplementedError
+            elif explainer_config.node_mask_type == MaskType.common_attributes:
+                node_feat_mask_ = self._reshape_common_attributes(
+                    node_mask, num_nodes)
```

Comment:
if `task_level==node` then wouldn't we want to zero out the mask of nodes outside the subgraph?

Reply:
Yes very good point, I'll update that ASAP.